zulip_bots/zulip_bots/test_lib.py

```diff
@@ -174,20 +181,36 @@ def mock_http_conversation(http_data):
     http_request = http_data.get('request')
     http_response = http_data.get('response')
     http_headers = http_data.get('response-headers')
-    with patch('requests.get') as mock_get:
-        mock_result = requests.Response()
-        mock_result._content = json.dumps(http_response).encode()  # type: ignore # We are modifying a "hidden" attribute.
-        mock_result.status_code = http_headers.get('status', 200)
-        mock_get.return_value = mock_result
-        yield
-        if 'params' in http_request:
-            params = http_request.get('params', None)
-            mock_get.assert_called_with(http_request['api_url'], params=params)
-        elif 'headers' in http_request:
-            headers = http_request.get('headers', None)
-            mock_get.assert_called_with(http_request['api_url'], headers=headers)
-        else:
-            mock_get.assert_called_with(http_request['api_url'])
+    http_method = http_request.get('method', 'GET')
+
+    if http_method == 'GET':
+        with patch('requests.get') as mock_get:
+            mock_get.return_value = get_response(http_response, http_headers)
+            yield
+            if 'params' in http_request:
+                params = http_request.get('params', None)
+                mock_get.assert_called_with(http_request['api_url'], params=params)
+            elif 'headers' in http_request:
+                headers = http_request.get('headers', None)
+                mock_get.assert_called_with(http_request['api_url'], headers=headers)
+            else:
+                mock_get.assert_called_with(http_request['api_url'])
+    else:
+        with patch('requests.post') as mock_post:
```

Comment:
The two codepaths in each `with` statement are almost identical. I actually think that the codepath for `mock_post` should also work with `mock_get`. Can you try de-duplicating code by putting it all into a function inside `mock_http_conversation` that is then called in each with statement?